### PR TITLE
mysqlcluster: Support ipv6 address. #441

### LINKED ISSUE #441: [feature] Support ipv6 address.
**Is your feature request related to a problem? Please describe.**

`127.0.0.1` in ipv6 is `::1`, Xenon May use `:: 1` to connect mysql instead of `127.0.0.1`, so we also need to provide the default account of `::1`.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by acekingke**
### What type of PR is this?

/bug
### Which issue(s) this PR fixes?

Fixes #441 

### What this PR does?

Summary:

### CODE REVIEW

File: mysqlcluster/syncer/statefulset.go [hunk 1]
```diff
@@ -427,6 +427,12 @@ func (s *StatefulSetSyncer) ensurePodSpec() corev1.PodSpec {
 		Affinity:           s.Spec.PodPolicy.Affinity,
 		PriorityClassName:  s.Spec.PodPolicy.PriorityClassName,
 		Tolerations:        s.Spec.PodPolicy.Tolerations,
+		HostAliases: []corev1.HostAlias{
```
Comment (zhl003) [inline mysqlcluster/syncer/statefulset.go:430]: HostAlias useless for this case
 default host has both 
 127.0.0.1 and ::1

File: sidecar/config.go [hunk 2]
```diff
@@ -333,7 +337,7 @@ func (cfg *Config) buildXenonConf() []byte {
 			"admin": "root",
 			"ping-timeout": %d,
 			"passwd": "%s",
-			"host": "localhost",
+			"host": "127.0.0.1",
```
Comment (zhl003) [inline sidecar/config.go:340]: root@127.0.0.1 should  set empty password

Comment (acekingke) [reply sidecar/config.go:340]: 127.0.0.1 and localhost with sample password now.

**Review by runkecheng — APPROVED** [review]

**Comment by runkecheng** [comment]
old code:
go-nyet sidecar/config.go 
sidecar/config.go:140:3:Shadowing variable `mysqlSemVer`

**Comment by zhl003** [comment]
LGTM.   
but in CRD:
```shell
                  rootHost: localhost
                  rootPassword: ""
```
should or not open to user for modify?  
if so， `rootHost` and `rootPassword` should  effective for root@'%'  ,but not root@localhost and root@127.0.0.1

**Review by zhl003 — COMMENTED** [review]
lgtm

File: sidecar/config.go [hunk 3]
```diff
@@ -348,7 +352,7 @@ func (cfg *Config) buildXenonConf() []byte {
 			"semi-sync-degrade": true,
 			"purge-binlog-disabled": true,
 			"super-idle": false
-		}
+		}  
 	}
```
Comment (andyli029) [inline sidecar/config.go:356]: Fix it.

File: sidecar/config.go [hunk 4]
```diff
@@ -348,7 +352,7 @@ func (cfg *Config) buildXenonConf() []byte {
 			"semi-sync-degrade": true,
 			"purge-binlog-disabled": true,
 			"super-idle": false
-		}
+		}
```
Comment (andyli029) [inline sidecar/config.go:355]: Fix it.

**Review by andyli029 — DISMISSED** [review]

**Review by andyli029 — APPROVED** [review]

**Pull request merged**